_posts/2019-01-05-its-time-to-use-spm.markdown

```diff
@@ -0,0 +1,167 @@
+---
+layout: epic
+title: It's time to use Swift Package Manager
+date: 2089-01-05
+author: [orta]
+categories: [swift, package-management, build tools, roads and bridges]
+comment_id: 517
+---
+
+It's been three years, and [Swift Package Manager][spm] (SPM) is at a point where it can be useful for iOS
+projects. It'll take a bit of sacrifice and a little bit of community spirit to fix some holes probably but **in my
+opinion, it's time for teams to start adopting SPM for their 3rd party dev tools**.
+
+**TDLR:** You should be using SPM for 3rd party dev tools like: [SwiftLint][sl], [SwiftFormat][sf], [Danger][ds],
+[Sourcery][srcy], [SwiftGen][sg] and [Git Hook management][kom].
+
+This post covers: What made it feasible to use SPM now? What are the downsides of the status quo? Why use SPM at
+all? What are the downsides to using SPM?
+
+<!-- more -->
+
+### What changed to make SPM usable?
+
+From my perspective, David Hart's [addition of `swift run`][swift-run] to SPM which shipped with Swift 4.0 is what
+pushed the project over the finish line to being useful for iOS developers. `swift run` is contextually the same as
+`bundle exec` in that it will run a locally bundled version of your executable.
+
+This means you can run `run swift swiftlint` and reliably get the same results as your CI and fellow developers.
```

Comment:
@orta Should this be `swift run swiftlint`?

Reply:
lolyes